.github/workflows/xbox.yml

```diff
@@ -13,20 +13,20 @@ jobs:
     steps:
     - name: Install and Setup Dependencies
       run: |
-        sudo apt-get update -y && sudo apt-get install -y flex bison clang lld
+        sudo apt-get update -y && sudo apt-get install -y flex bison clang lld llvm
         cd ~
-        git clone https://github.com/XboxDev/nxdk.git --recursive
- 
+        git clone --depth 1 --recurse-submodules --shallow-submodules https://github.com/XboxDev/nxdk.git
+
     - name: Checkout Repository
       uses: actions/checkout@v2
       with:
-        submodules: true
+        submodules: recursive
 
     - name: Compile
+      working-directory: example
       run: |
-        export PATH=$PATH:/usr/lib/llvm-10/bin
-        cd example/
-        make -f Makefile.nxdk -j$(nproc) NXDK_DIR=~/nxdk
+        eval $(~/nxdk/bin/activate -s)
```

Comment:
This is stupid. It will put `activate` in the PATH and that will potentially wreak havoc (`activate` is way too generic for a tool name and not clear that this will be nxdk); it should have been outside of bin/ or be named nxdk-activate.
This will affect CI scripts, and I'd recommend to address this before this usage style spreads more widely.

Reply:
Hi @JayFoxRox, long time no see.

I'm not entirely sure how it will cause havoc when it has nxdk in the path name. If this is an issue, then better file a ticket in nxdk repo. Currently, any nxdk projects that are cloning up-to-date nxdk repo are affected for GitHub CI. Unless workflows are updated to use NXDK's new build system as requirement to function properly.

Reply:
I see the problem from inside `activate` script you're talking about. Only way to resolve this is submit a fix in nxdk repository. Then re-update affecting repositories.

Reply:
> I'm not entirely sure how it will cause havoc when it has nxdk in the path name.

It's more of a problem for users on the CLI, but it will change how CI interacts with nxdk.

> Only way to resolve this is submit a fix in nxdk repository. Then re-update affecting repositories.

Yes. It's not a problem in this PR. However, the upstream should be fixed before this lands in more CI scripts and the existing CI scripts will have to be changed again in the future if the fix lands upstream (so it should be done asap).